### DISCUSSION TITLE
Creating .geo file for eeg cap

### DISCUSSION THREAD

**Discussion by michefabb** [simnibs/simnibs#468] [7.Creating head models] [answered]
Dear all,

I have created the .csv file of my customized EEG cap for my subject, how can I now create the .geo file from this?

Appreciated, thanks,

Mic

**Comment by jduemose** [comment]
Hi Mic

You can use the function `simnibs.mesh_tools.mesh_io.write_geo_spheres` which takes positions and a filename as input, see https://github.com/simnibs/simnibs/blob/d045e8ccda6b349cbec639563af32dac69e95909/simnibs/mesh_tools/mesh_io.py#L6058

Best,
Jesper

**Reply by michefabb** [reply]
Dear @jduemose ,

thanks for your reply, I have the csv file that contains:
Fiducial,2.7245,103.0464,-32.0877,NZ
Fiducial,-58.6382,8.3215,-30.5282,LPA
Fiducial,75.321,16.6186,-32.8507,RPA
Electrode,69.2533,78.3299,-16.3391,F10
Electrode,65.5096,86.0913,18.4619,AF8
............

I see the geo files created by `charm` have the following syntax:
View""{
SP(47.831055654279794, -60.76859562761443, 63.465831140148076){0};
T3(47.831055654279794, -60.76859562761443, 68.46583114014808, 0){"P2"};
SP(16.969719019789537, -82.84691899988316, 42.745748710986355){0};
T3(16.969719019789537, -82.84691899988316, 47.745748710986355, 0){"POz"};
..............
};
myView = PostProcessing.NbViews-1;
View[myView].PointType=1;
View[myView].PointSize=6;
View[myView].LineType=1;
View[myView].LineWidth=2;

I am unsure how to use the `write_geo_spheres` python command in order to create a similar geo file from the csv: would you kindly provide an example?

Thanks, appreciated,

Mic

**Reply by jduemose** [reply, marked as answer]
It turns out there is this function `simnibs.utils.transformations.csv_to_geo` which does exactly what you want. I just didn't know about it until now :) 
https://github.com/simnibs/simnibs/blob/d045e8ccda6b349cbec639563af32dac69e95909/simnibs/utils/transformations.py#L1222

**Reply by michefabb** [reply]
Thanks @jduemose , this is what I was looking for.
Let me just add a bug report: it looks it does not consider the electrode in the first row (i.e. it starts from the second row), maybe something to look into if you wish.

Appreciated!

Mic

**Reply by TorgeW** [reply]
https://github.com/simnibs/simnibs/blob/d045e8ccda6b349cbec639563af32dac69e95909/simnibs/utils/csv_reader.py#L9

Seams like your first row is detected as a header.

**Reply by michefabb** [reply]
correct @TorgeW but as you can see it is exactly like all the others...maybe a bug?

**Reply by TorgeW** [reply]
I put the 5 positions you shared into a csv file and ran csv_to_geo and it worked for me. 
I get all 5 positions in the geo file and they all also show in gmsh.

**Reply by michefabb** [reply]
Thanks @TorgeW ,

in that case I do not have an explanation for this...however it is a minor, I corrected it adding a header and getting it fixed!

Mic
